set.mm

```diff
@@ -757432,6 +757435,319 @@ last digit of the integer part (for ` b = ` 10 the first digit before
   $}
 
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  N-ary functions
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+
+  According to Wikipedia ("Arity", ~ https://en.wikipedia.org/wiki/Arity ,
+  19-May-2024): "In logic, mathematics, and computer science, _arity_ is the
+  number of arguments or operands taken by a function, operation or relation."
+
+  N-ary functions are often also called multivariate functions, without
+  indicating the actual number of argumens. See also Wikipedia
+  ("Multivariate functions",   19-May-2024,
+  ~ https://en.wikipedia.org/wiki/Function_(mathematics)#Multivariate_functions
+  ): "A _multivariate function_, multivariable function, or function of several
+  variables is a function that depends on several arguments.  ...  Formally, a
+  function of n variables is a function whose domain is a set of n-tuples.
+  For example, multiplication of integers is a function of two variables, or
+  _bivariate function_, whose domain is the set of all ordered pairs (2-tuples)
+  of integers, and whose codomain is the set of integers.  The same is true for
+  every binary operation.  Commonly, an n-tuple is denoted enclosed between
+  parentheses, such as in ( 1 , 2 , ... , n ).  When using functional notation,
+  one usually omits the parentheses surrounding tuples, writing
+  f ( x_1 , ... , x_n ) instead of f ( ( x_1 , ... , x_n ) ).  Given n sets
+  X_1 , ... , X_n , the set of all n-tuples ( x_1 , ... , x_n ) such that
+  x_1 is element of X_1 , ... , x_n is element of X_n is called the Cartesian
+  product of X_1 , ... , X_n , and denoted X_1 X ... X X_n .
```

Comment:
Another interesting point is that because of the way things are defined in set.mm, a `2 N-ary` function is not exactly the same thing as an operation. The domain of the former (this function) is the set of words of length 2 (functions from `( 0 ..^ 2 )` to `X` ), while the domain of the latter is set of pairs of elements of `X`, i.e. the cartesian product `( X X. X )`. The same is true for univariate functions. 

It can be noted though that those are equinumerous, see `~naryenef`.

Reply:
In a next step, I will investigate 2-ary (i.e, binary) functions/Operations in more detail.